exporters/metrics/src/main/java/com.google.cloud.opentelemetry.metric/MetricTranslator.java

```diff
@@ -0,0 +1,234 @@
+          .build())
+      .put("gke_container", ImmutableMap.<String, String>builder()
+          .put("k8s.cluster.name", GCP_CLUSTER_NAME_KEY)
+          .put("k8s.namespace.name", GCP_NAMESPACE_KEY)
+          .put("k8s.pod.name", GCP_POD_KEY)
+          .put("host.id", GCP_HOST_ID_KEY)
+          .put("container.name", GCP_CONTAINER_KEY)
+          .put("cloud.account.id", GCP_ACCOUNT_ID_KEY)
+          .put("cloud.zone", GCP_ZONE_KEY)
+          .build())
+      .build();
+
+
+  static Metric mapMetric(MetricData metric, String type, String uniqueIdentifier) {
+    Metric.Builder metricBuilder = Metric.newBuilder().setType(type);
+    metric.getDescriptor().getConstantLabels().forEach(metricBuilder::putLabels);
+    if (uniqueIdentifier != null) {
+      metricBuilder.putLabels(UNIQUE_IDENTIFIER_KEY, uniqueIdentifier);
+    }
+    return metricBuilder.build();
+  }
+
+  static MetricDescriptor mapMetricDescriptor(MetricData metric, String uniqueIdentifier) {
+    String instrumentName = metric.getInstrumentationLibraryInfo().getName();
+    MetricDescriptor.Builder builder = MetricDescriptor.newBuilder().setDisplayName(instrumentName)
+        .setType(mapMetricType(instrumentName));
+    metric.getDescriptor().getConstantLabels().forEach((key, value) -> builder.addLabels(mapConstantLabel(key, value)));
+    if (uniqueIdentifier != null) {
+      builder.addLabels(
+          LabelDescriptor.newBuilder().setKey(UNIQUE_IDENTIFIER_KEY).setValueType(LabelDescriptor.ValueType.STRING)
+              .build());
+    }
+
+    MetricData.Descriptor.Type metricType = metric.getDescriptor().getType();
+    if (GAUGE_TYPES.contains(metricType)) {
+      builder.setMetricKind(MetricDescriptor.MetricKind.GAUGE);
+    } else if (CUMULATIVE_TYPES.contains(metricType)) {
+      builder.setMetricKind(MetricDescriptor.MetricKind.CUMULATIVE);
+    } else {
+      logger.error("Metric type {} not supported", metricType);
+      return null;
+    }
+    if (LONG_TYPES.contains(metricType)) {
+      builder.setValueType(MetricDescriptor.ValueType.INT64);
+    } else if (DOUBLE_TYPES.contains(metricType)) {
+      builder.setValueType(MetricDescriptor.ValueType.DOUBLE);
+    } else {
+      logger.error("Metric type {} not supported", metricType);
+      return null;
+    }
+    return builder.build();
+  }
+
+  private static String mapMetricType(String instrumentName) {
+    for (String domain : KNOWN_DOMAINS) {
+      if (instrumentName.contains(domain)) {
+        return instrumentName;
+      }
+    }
+    return DESCRIPTOR_TYPE_URL + instrumentName;
+  }
+
+  static MonitoredResource mapGcpMonitoredResource(Resource resource) {
+    ReadableAttributes attributes = resource.getAttributes();
+    if (attributes.get("cloud.provider") == null ||
+        !attributes.get("cloud.provider").getStringValue().equals("gcp")) {
+      return null;
+    }
+    String resourceType = attributes.get("gcp.resource_type").getStringValue();
```

Comment:
@james-bebbington @aabmass have we agreed to keep this attribute, or should we switch to [semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/tree/master/specification/resource/semantic_conventions)?

I would suggest to just remove resource GCP mapping for now from the code, and add it later.

Reply:
To clarify, the question was whether we should infer resource type based on other attributes, like we did in https://github.com/open-telemetry/opentelemetry-collector/pull/1462